ee_tests/src/specs/page_objects/app.page.ts

```diff
@@ -13,10 +13,10 @@ export abstract class AppPage extends BasePage {
   /**
    * Extend this class, to describe Application Page(after logging in)
    *
-   * @param {url} elementFinder ElementFinder that you want to extend
+   * @param {url} string URL where the extended page resides
    */
-  constructor() {
-    super();
+  constructor(url?: string) {
```

Comment:
Please note the comments about url? in base.page.ts. URL set to '' is a valid one.

Why was this change needed by the way? Can this be made into a separate commit but part of the same PR?

Reply:
The parent class `base.page's` constructor accepts a url/undefined value.
Using `url?: string` would ensure user passes only a string and not a boolean/number.
Also, if the user doesn't pass anything `url` would be undefined, as we expect it to be.